### DISCUSSION TITLE
How to convert arrayfire array in Rust from f64 to i32?

### DISCUSSION THREAD

**Discussion by 3togo** [arrayfire/arrayfire-rust#340] [Q&A] [answered]
To convert arrayfire array in C++ from f64 to i32, we could simply use as(i32). What is the equivalent method in rust version?

// Example, double precision array
af::array A_double(100, 1, f64); // f64 = double precision

// Cast to an i32 array
af::array A_single = A_double.as(i32);

**Comment by 3togo** [comment, marked as answer]
I guess it should be 
A_double.cast::<i32>();

**Reply by 9prady9** [reply]
Docs - https://arrayfire.org/arrayfire-rust/arrayfire/struct.Array.html#method.cast
Example usage - https://github.com/arrayfire/arrayfire-rust/blob/master/examples/conway.rs#L27
